alpakaConfig.cmake

```diff
@@ -339,12 +339,23 @@ IF(ALPAKA_ACC_GPU_CUDA_ENABLE)
                     LIST(APPEND CUDA_NVCC_FLAGS "-Wno-deprecated-gpu-targets")
                 ENDIF()
 
-                IF(NOT CUDA_VERSION VERSION_LESS 7.5)
+                IF(CUDA_VERSION VERSION_LESS 7.5)
+                    # CUDA 7.0
+                    LIST(APPEND CUDA_NVCC_FLAGS "--relaxed-constexpr")
+                ELSEIF(CUDA_VERSION VERSION_LESS 8.0)
+                    # CUDA 7.5
                     LIST(APPEND CUDA_NVCC_FLAGS "--expt-extended-lambda")
                     LIST(APPEND CUDA_NVCC_FLAGS "--expt-relaxed-constexpr")
+                ELSEIF(CUDA_VERSION VERSION_LESS 9.0)
+                    # CUDA 8.0
+                    LIST(APPEND CUDA_NVCC_FLAGS "--expt-extended-lambda")
+                    IF((NOT CMAKE_COMPILER_IS_GNUCXX) OR (CMAKE_COMPILER_IS_GNUCXX AND CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL 5))
```

Comment:
note: `--expt-relaxed-constexp` is available with gcc 5.X. 
Also it is not possible to bring this mainline else we lose that the feature that we can write the code once and run it on different platforms and compiler without `#ifdef` within the user code.

Reply:
NVCC 8.0 in combination with gcc 5.4 triggers the following compiler error in all tests:
```C++
/usr/include/c++/5/tuple(484) (col. 17): error: calling a __host__ function("std::_Tuple_impl<(unsigned long)0ul,  ::std::_Bind< ::std::_Mem_fn<void (     ::alpaka::core::detail::ConcurrentExecPool<unsigned long,  ::std::thread,  ::std::promise, void,  ::std::mutex,  ::std::condition_variable, (bool)0> ::*)()>  ( ::alpaka::core::detail::ConcurrentExecPool<unsigned long,  ::std::thread,  ::std::promise, void,  ::std::mutex,  ::std::condition_variable, (bool)0>  *)>  > ::_Tuple_impl [subobject]") from a __device__ function("std::tuple< ::std::_Bind< ::std::_Mem_fn<void (     ::alpaka::core::detail::ConcurrentExecPool<unsigned long,  ::std::thread,  ::std::promise, void,  ::std::mutex,  ::std::condition_variable, (bool)0> ::*)()>  ( ::alpaka::core::detail::ConcurrentExecPool<unsigned long,  ::std::thread,  ::std::promise, void,  ::std::mutex,  ::std::condition_variable, (bool)0>  *)>  > ::tuple") is not allowed
```
So for this compiler combination the CUDA backend might be incompatible with the std::thread and fiber backends. I will try.

Reply:
It looks like https://github.com/ComputationalRadiationPhysics/alpaka/pull/416

Reply:
I suppose that the error occurs because gcc 5.4 marked many more STL functions as `constepr`. However, this does not seem to be compatible with nvcc 8.0 and the functions it is replacing/overriding e.g.
`/usr/include/c++/5/cmath(362): error: calling a __host__ function("__builtin_logl") from a __device__ function("std::log") is not allowed`

I was able to replace the code that used `std::bind` with lambdas so that the CPU backends are functional again. The only two remaining errors are within the `rand` and `stream` unit tests. I will look for a workaround for the `stream` test failure but the CPU random number distributions based on the STL seem not be compatible with this compiler because of their usage of `std::log`.